packages/async-rewriter3/src/lib.rs

```diff
@@ -0,0 +1,400 @@
+use std::{borrow::Borrow, collections::VecDeque};
+use wasm_bindgen::prelude::*;
+use rslint_parser::{ast::{ArrowExpr, AssignExpr, CallExpr, ClassDecl, Expr, ExprOrBlock, ExprStmt, FnDecl, FnExpr, ObjectPatternProp, Pattern, PropName, ReturnStmt, UnaryExpr, VarDecl}, parse_text, AstNode, SyntaxNode, TextSize};
+
+#[derive(Debug)]
+enum InsertionText {
+    Static(&'static str),
+    Dynamic(String)
+}
+
+#[derive(Debug)]
+struct Insertion {
+    offset: TextSize,
+    text: InsertionText,
+    original_ordering: Option<u32>
+}
+
+impl Insertion {
+    pub const fn new(offset: TextSize, text: &'static str) -> Insertion {
+        Insertion {
+            offset,
+            text: InsertionText::Static(text),
+            original_ordering: None
+        }
+    }
+
+    pub const fn new_dynamic(offset: TextSize, text: String) -> Insertion {
+        Insertion {
+            offset,
+            text: InsertionText::Dynamic(text),
+            original_ordering: None
+        }
+    }
+}
+
+struct InsertionList {
+    list: VecDeque<Insertion>,
+    vars: Vec<String>
+}
+
+impl InsertionList {
+    pub const fn new() -> InsertionList {
+        InsertionList {
+            list: VecDeque::new(),
+            vars: Vec::new()
+        }
+    }
+
+    fn append(&mut self, other: &mut Self) {
+        self.list.append(&mut other.list);
+        self.vars.append(&mut other.vars);
+    }
+
+    fn push_back(&mut self, insertion: Insertion) {
+        self.list.push_back(insertion);
+    }
+
+    fn pop_back(&mut self) {
+        self.list.pop_back();
+    }
+
+    fn add_variable(&mut self, variable: String) {
+        self.vars.push(variable);
+    }
+}
+
+fn is_block(body: &ExprOrBlock) -> bool {
+    match body {
```

Comment:
If you want to make it slimmer (just for fun) you can use matches!

It would be like (with typos for sure):


```rs
matches!(body, ExprOrBlock::Block(_))
```